src/SDKs/CognitiveServices/dataPlane/Search/BingVisualSearch/BingVisualSearch.Tests/VisualSearchTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Azure.CognitiveServices.Search.VisualSearch;
+using Microsoft.Azure.CognitiveServices.Search.VisualSearch.Models;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Newtonsoft.Json;
+using Xunit;
+using System.IO;
+
+namespace SearchSDK.Tests
+{
+    public class VisualSearchTests
+    {
+        private const string ImageInsightsToken = "bcid_2B63103C3B473829DCC4F03074E157E6*ccid_sGbXtPJL*mid_3B429ACA76F93D5417A3F0D646E96520CE704B3F*simid_608000193735756642";
+        private const string ImageUrl = "http://3.bp.blogspot.com/-dbDvMyLyZUU/T2fvDTnPFzI/AAAAAAAAAxs/i53ZHxUtZRo/s400/915534489_de793b90b5_b.jpg";
+        private static string SubscriptionKey = "fake";
+        CropArea CropArea = new CropArea(top: (float)0.1, bottom: (float)0.5, left: (float)0.1, right: (float)0.9);
+
+        [Fact]
+        public void VisualSearchWithBinary()
+        {
+            using (MockContext context = MockContext.Start(this.GetType().FullName))
+            {
+                HttpMockServer.Initialize(this.GetType().FullName, "VisualSearchWithBinary");
+
+                IVisualSearchAPI client = new VisualSearchAPI(new ApiKeyServiceClientCredentials(SubscriptionKey), HttpMockServer.CreateInstance());
+
+                using (FileStream stream = new FileStream(Path.Combine("TestImages", "image.jpg"), FileMode.Open))
+                {
+                    KnowledgeRequest KnowledgeRequest = new KnowledgeRequest();
+
+                    var resp = client.Images.VisualSearchMethodAsync(image: stream, knowledgeRequest: JsonConvert.SerializeObject(KnowledgeRequest)).Result;
```

Comment:
@shakao why are you not using the new function from your customization layer.
Why KnowledgeRequest is still being serialized?

Reply:
the new test is "VisualSearchWithKnowledgeRequestObject" (line 102)! these tests are there for the serialized functions

Reply:
and the extensions are in the /Customizations folder (IImages.cs, Images.cs, ImageExtensions.cs)